components/p3a/p3a_service.cc

```diff
@@ -236,18 +229,29 @@ void P3AService::OnHistogramChanged(const char* histogram_name,
     return;
   }
 
-  // Note that we store only buckets, not actual values.
-  size_t bucket = 0u;
-  const bool ok = samples->Iterator()->GetBucketIndex(&bucket);
-  if (!ok) {
-    LOG(ERROR) << "Only linear histograms are supported at the moment!";
-    NOTREACHED();
-    return;
-  }
+  // Assume that the sample == bucket for P3A measurements.
+  size_t bucket = std::max(sample, 0);
 
   // Special handling of P2A histograms.
   if (base::StartsWith(histogram_name, "Brave.P2A",
                        base::CompareCase::SENSITIVE)) {
+    std::unique_ptr<base::HistogramSamples> samples =
+        base::StatisticsRecorder::FindHistogram(histogram_name)
+            ->SnapshotDelta();
+
+    // Stop now if there's nothing to do.
+    if (samples->Iterator()->Done()) {
+      return;
+    }
+
+    // Note that we store only buckets, not actual values.
+    const bool ok = samples->Iterator()->GetBucketIndex(&bucket);
```

Comment:
can you remind me, why the above logic (i.e. using sample as a bucket) doesn't work for P2A ?

Reply:
I'm not sure, since I'm not too familiar with P2A. I left the functionality as is, to reduce any risk. Also, I believe there is an effort to deprecate P2A if I'm not mistaken